### PR TITLE
options/posix: implement putpwent

### PR DESCRIPTION
**Pull request by Mathewnd**
Implement stubbed function.

### CODE REVIEW

File: options/posix/generic/pwd-stubs.cpp [hunk 1]
```diff
@@ -273,9 +273,18 @@ void endpwent(void) {
 	close_global_file();
 }
 
-int putpwent(const struct passwd *, FILE *) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+static bool invalid(const char *s) {
```
Comment (Geertiebear) [inline options/posix/generic/pwd-stubs.cpp:276]: Could you make this a local lambda in the function please? Avoids littering the global namespace with an unspecific name like ``invalid``

**Review by Geertiebear — CHANGES_REQUESTED** [review]

File: options/posix/generic/pwd-stubs.cpp (hunk 1 above)
Comment (Mathewnd) [reply options/posix/generic/pwd-stubs.cpp]: done

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Review by Geertiebear — APPROVED** [review]
LGTM

**Pull request merged**